src/main/java/uk/gov/cshr/civilservant/controller/CivilServantController.java

```diff
@@ -105,6 +114,83 @@ public CivilServantController(LineManagerService lineManagerService, CivilServan
         return ResponseEntity.unprocessableEntity().build();
     }
 
+    @GetMapping("/org")
+    public ResponseEntity<OrgCodeDTO> getOrgCodeForCivilServant(@RequestParam(value = "uid") String uid) {
+        log.debug("Getting civil servant org details for user with uid " + uid);
+
+        Optional<CivilServant> civilServant = civilServantRepository.findByIdentity(uid);
+
+        if (civilServant.isPresent()) {
+            return civilServantResourceFactory.getCivilServantOrganisationalUnitCode(civilServant.get())
+                    .map(orgCodeDTO -> ResponseEntity.ok(orgCodeDTO))
+                    .orElse(buildNotFound(uid));
+        } else {
+            return buildNotFound(uid);
+        }
+
+    }
+
+    private ResponseEntity<OrgCodeDTO> buildNotFound(String uid) {
+        log.warn(String.format("Civil Servant with uid %s not found", uid));
+        return ResponseEntity.notFound().build();
+    }
+
+    @PatchMapping("/org")
+    @PreAuthorize("isAuthenticated()")
+    public ResponseEntity updateOrganisation(@Valid @RequestBody UpdateOrganisationDTO updateOrganisationDTO) {
+
+        log.info("updating civil servants organisation for organisation=" + updateOrganisationDTO.getOrganisation());
+
+        try {
+            Optional<CivilServant> optionalCivilServant = civilServantRepository.findByPrincipal();
+            if (optionalCivilServant.isPresent()) {
```

Comment:
Replacing your `Optional.isPresent()` and `Optional.get()` with one of the following allows you a more flexible method of handling conditional logic. 
```
Optional.isPresent(obj -> {})
Optional.map(obj -> {})
Optional.orElseGet(obj -> {})
```

Reply:
Updated.